### PR TITLE
feat:add SQLServerTableMetaDataLoader

### PR DESCRIPTION
**Pull request by totalo**
For #9617 .

Changes proposed in this pull request:
- add SQLServerTableMetaDataLoader

### CODE REVIEW

**Review by terrymanu — CHANGES_REQUESTED** [review]
The CI is broken, please fix it

**Comment by totalo** [comment]
> The CI is broken, please fix it

I have corrected the related errors, please review again

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/SQLServerTableMetaDataLoader.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Table meta data loader for SQLServer.
+ * @author totalo
+ * @date 20210316
```
Comment (terrymanu) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/SQLServerTableMetaDataLoader.java:42]: Please remove @author and @date, let's keep consist with other codes

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/SQLServerTableMetaDataLoader.java (hunk 1 above)
Comment (totalo) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/builder/loader/dialect/SQLServerTableMetaDataLoader.java]: ok ~

File: shardingsphere-scaling/shardingsphere-scaling-bootstrap/src/test/java/org/apache/shardingsphere/scaling/web/HttpServerInitializerTest.java [hunk 2]
```diff
@@ -45,9 +42,8 @@
     private ChannelPipeline channelPipeline;
     
     @Before
-    @SneakyThrows(ReflectiveOperationException.class)
     public void setUp() {
-        ReflectionUtil.setFieldValue(ScalingContext.getInstance(), "serverConfig", new ServerConfiguration());
+        ServerConfigurationInitializer.init();
```
Comment (terrymanu) [inline shardingsphere-scaling/shardingsphere-scaling-bootstrap/src/test/java/org/apache/shardingsphere/scaling/web/HttpServerInitializerTest.java:46]: This change seem no relate with your PR, why to change it?

File: shardingsphere-proxy/shardingsphere-proxy-bootstrap/src/test/java/org/apache/shardingsphere/proxy/initializer/impl/GovernanceBootstrapInitializerTest.java [hunk 3]
```diff
@@ -227,4 +227,10 @@ public void assertDecorateTransactionContexts() {
     protected void prepareSpecifiedInitializer() {
         setInitializer(new GovernanceBootstrapInitializer());
     }
+
+    private StandardMetaDataContexts mockStandardMetaDataContexts() {
+        StandardMetaDataContexts standardMetaDataContexts = mock(StandardMetaDataContexts.class);
+        when(standardMetaDataContexts.getProps()).thenReturn(new ConfigurationProperties(new Properties()));
+        return standardMetaDataContexts;
+    }
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-bootstrap/src/test/java/org/apache/shardingsphere/proxy/initializer/impl/GovernanceBootstrapInitializerTest.java:235]: Another PR fix NPE, so, could you revert the changes?

**Review by terrymanu — CHANGES_REQUESTED** [review]
Please revert other changes beside SQLServer meta data loader

File: shardingsphere-proxy/shardingsphere-proxy-bootstrap/src/test/java/org/apache/shardingsphere/proxy/initializer/impl/GovernanceBootstrapInitializerTest.java (hunk 3 above)
Comment (totalo) [reply shardingsphere-proxy/shardingsphere-proxy-bootstrap/src/test/java/org/apache/shardingsphere/proxy/initializer/impl/GovernanceBootstrapInitializerTest.java:235]: ok~

File: shardingsphere-scaling/shardingsphere-scaling-bootstrap/src/test/java/org/apache/shardingsphere/scaling/web/HttpServerInitializerTest.java (hunk 2 above)
Comment (totalo) [reply shardingsphere-scaling/shardingsphere-scaling-bootstrap/src/test/java/org/apache/shardingsphere/scaling/web/HttpServerInitializerTest.java:46]: Because in my local test, I found that the serverConfig is empty. When instantiating the HttpServerHandler, the checkServerConfig method can’t get through.

**Comment by totalo** [comment]
> Please revert other changes beside SQLServer meta data loader

ok i hava done

**Review by Lucas-307 — APPROVED** [review]

**Review by terrymanu — APPROVED** [review]
